File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/repository/transformer/RxGistCommitTransformer.java
package ru.avystavkin.githubgists.repository.transformer;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import ru.avystavkin.githubgists.models.local.GistCommit;
import ru.avystavkin.githubgists.models.server.GistCommitServer;

public class RxGistCommitTransformer implements ObservableTransformer<List<GistCommitServer>, List<GistCommit>> {


    @Override
    public ObservableSource<List<GistCommit>> apply(Observable<List<GistCommitServer>> observable) {
        return observable
                .flatMap( Observable::fromIterable)
                .map(TransformerHelper::convertGistCommitsHistory)
                .toList()
                .flatMapObservable(Observable::fromArray);
    }
}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/repository/github/transformer/RxGistCommitTransformer.kt
package ru.avystavkin.githubgists.repository.github.transformer

import io.reactivex.Observable
import io.reactivex.ObservableSource
import io.reactivex.ObservableTransformer
import ru.avystavkin.githubgists.models.database.GistCommit
import ru.avystavkin.githubgists.models.server.GistCommitServer

class RxGistCommitTransformer : ObservableTransformer<List<GistCommitServer>, List<GistCommit>> {

    override fun apply(observable: Observable<List<GistCommitServer>>): ObservableSource<List<GistCommit>> {
        return observable
                .flatMap { x -> Observable.fromIterable(x) }
                .map { x -> TransformerHelper.convertGistCommitsHistory(x) }
                .toList()
                .flatMapObservable { x -> Observable.fromArray(x) }
    }
}
